### PR TITLE
Support different node flags for different versions of node.

### PR DESCRIPTION
**Pull request by sbc100**
This is mostly about what flags to pass to node during testing, but also applies to `CMAKE_CROSSCOMPILING_EMULATOR` and to the #! used in autoconf tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,10 @@ orbs:
 executors:
   linux-node:
     docker:
-      - image: circleci/node:stretch
+      - image: cimg/node:19.0.0
```
Comment (kripken) [inline .circleci/config.yml:9]: Do we need both the docker image for node 19 and also to download it manually below?

File: tools/shared.py [hunk 2]
```diff
@@ -336,11 +337,34 @@ def check_node_version():
...
     expected = '.'.join(str(v) for v in MINIMUM_NODE_VERSION)
     diagnostics.warning('version-check', f'node version appears too old (seeing "{actual}", expected "v{expected}")')
 
+  return version
+
+
+def node_bigint_flags():
+  node_version = check_node_version()
+  # wasm bigint was enabled by default in node v16.
+  # prior to that the --experimental-wasm-bigint flag is required.
```
Comment (kripken) [inline tools/shared.py:352]:
```suggestion
```

File: tools/shared.py [hunk 3]
```diff
@@ -336,11 +337,34 @@ def check_node_version():
...
+  if node_version and node_version < (16, 0, 0):
+    return ['--experimental-wasm-bigint']
+  else:
+    return []
+
+
+def node_pthread_flags():
+  node_version = check_node_version()
+  # bulk memory and wasm threads were enabled by default in node v16.
+  # prior to that the --experimental-wasm-bulk-memory flag is required.
```
Comment (kripken) [inline tools/shared.py:362]:
```suggestion
```

Comment (kripken) [reply tools/shared.py:362]: This duplicates the code and seems unnecessary. (It was also missing one flag unless that was intentional?)

**Review by kripken — APPROVED** [review]
Nice!

File: .circleci/config.yml (hunk 1 above)
Comment (sbc100) [reply .circleci/config.yml:9]: This is actually a separate change, I think i'll just revert it.  we don't run these new tests in the `linux-node` image.

**Review by aheejin — APPROVED** [review]

**Review by kripken — APPROVED** [review]

**Pull request merged**